rbm_softmax.py

```diff
@@ -0,0 +1,504 @@
+"""Restricted Boltzmann Machine with softmax visible units.
```

Comment:
I think it's worthwhile to be able to see the diff. If you're as keen as I, I think the following three commits will do the trick:

1. delete `rbm_softmax.py`
2. add the one from sklearn
3. replace it with your version